WooCommerce/Classes/ViewRelated/Dashboard/Settings/SettingsViewController.swift

```diff
@@ -112,35 +124,35 @@ private extension SettingsViewController {
         tableView.delegate = self
     }
 
-    func refreshResultsController() {
-        try? resultsController.performFetch()
-        sites = resultsController.fetchedObjects
+    func configureResultsControllers(onReload: @escaping () -> Void) {
+        configureSiteResultsController(onReload: onReload)
+        configureAccountResultsController(onReload: onReload)
     }
 
-    func checkAvailabilityForPayments(onCompletion: @escaping () -> Void) {
-        guard let siteID = self.siteID else {
-            canCollectPayments = false
-            onCompletion()
+    private func configureSiteResultsController(onReload: @escaping () -> Void) {
+        try? sitesResultsController.performFetch()
+        sites = sitesResultsController.fetchedObjects
+    }
+
+    private func configureAccountResultsController(onReload: @escaping () -> Void) {
+        /// Bail early if the card present payments feature is disabled
+        ///
+        guard ServiceLocator.featureFlagService.isFeatureFlagEnabled(.cardPresentPayments) else {
             return
         }
 
-        let action = WCPayAction.loadAccount(siteID: siteID) { [weak self] result in
+        gatewayAccountResultsController.onDidChangeContent = {
+            onReload()
+        }
+
+        gatewayAccountResultsController.onDidResetContent = { [weak self] in
             guard let self = self else {
                 return
             }
-
-            switch result {
-            case .failure:
-                self.canCollectPayments = false
-            case .success(let account):
-                self.canCollectPayments = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.cardPresentPayments) &&
-                    account.isCardPresentEligible
-            }
-
-            onCompletion()
+            onReload()
         }
 
-        ServiceLocator.stores.dispatch(action)
+        try? gatewayAccountResultsController.performFetch()
```

Comment:
We'd probably need to do something like:
`gatewayAccounts = gatewayAccountResultsController.fetchedObjects`

Reply:
In https://github.com/woocommerce/woocommerce-ios/pull/4381/commits/775566b5dcc26310f255e389af0a617e923a744c I removed this change - I'm deferring view controller changes to the last pull in this series.